tests/transition.md

```diff
@@ -659,3 +662,174 @@ All test cases described in this document are beginning in a pre-Merge world and
+* `TERMINAL_BLOCK_HASH` – set to the hash of a certain block to become the terminal PoW block.
+* `TERMINAL_BLOCK_HASH_ACTIVATION_EPOCH` - set to the epoch when the Merge transition will be activated.
+
+
+### EL client tests
+
+EL client tests for `TBH` (`TERMINAL_BLOCK_HASH`) override are design with the following *potential* implementation in mind:
+* EL always triggers the Merge upgrade upon reaching `TERMINAL_TOTAL_DIFFICULTY`
+* `TERMINAL_BLOCK_HASH` + `TERMINAL_BLOCK_NUMBER` whitelists the canonical chain
+
+Effectively, this semantics gives the desired result. But if EL has two chains that have reached `TTD`, EL may not reject a transition payload built atop of a block which `blockHash != TBH`. But we have CL side to enforce the terminal block's `blockHash == TBH`. Which means that ommitting this check is OK on EL side. Tests in this section *doesn't* assume that `blockHash == TBH` is enforced by EL client implementations.
+
+With the above thoughts in mind, the goal of this section is to check that `TBH` override doesn't affect `TTD` block condition.
+
+* [ ] Propose valid transition block with enabled `TBH` override
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * `Genesis <- ... <- TB`, `TB` is a valid terminal block, `TB.blockHash == TBH`
+  * `B: Genesis <- ... <- Bn`, `B` chain is heavier than `A` and would be canonical unless `TBH` override is activated
+  * EL starts with `A` and `B` imported
+  * `forkchoiceUpdated(headBlockHash: TB.blockHash, payloadAttributes: mergeTransitionBlockAttributes)`
+    * EL's head is set to `headBlockHash`
+    * EL returns `{status: VALID, payloadId: mergeTransitionPayloadId}`
+  * `getPayload(mergeTransitionPayloadId)`
+    * EL returns merge transition payload
+  * `newPayload(mergeTransitionPayload)`
+    * EL returns `VALID`
+  
+  </details>
+
+* [ ] Build atop of invalid terminal block with enabled `TBH` override
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * `Genesis <- ... <- INV_TB`, `INV_TB` is a *valid PoW* but an *invalid terminal* block, i.e. `INV_TB.TD < TTD`
+  * EL starts with `INV_TB` as the head
+  * `forkchoiceUpdated(headBlockHash: TB.blockHash, payloadAttributes: mergeTransitionBlockAttributes)`
+    * EL returns `{status: INVALID, latestValidHash: 0x00..00, payloadId: null}`
+    * EL's head points to `INV_TB` if `INV_TB.TD < TTD`, and to `INV_TB.parent` if `INV_TB.parent.TD >= TTD`
+  
+  </details>
+
+* [ ] Transition on a valid chain with enabled `TBH` override
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * `A: Genesis <- ... <- TB <- P1 <- P2 <- P3`, `TB` is a valid terminal block, i.e. `TB.blockHash == TBH`
+  * `B: Genesis <- ... <- Bn`, `B` chain is heavier than `A` and would be canonical unless `TBH` override is activated
+  * EL starts with `A` and `B` imported
```

Comment:
Is `A` imported up to `TB` or are `P1`, `P2` and `P3` also imported from the very beginning ?